xbmc/filesystem/ShoutcastFile.cpp

```diff
@@ -163,12 +167,28 @@ bool CShoutcastFile::ExtractTagInfo(const char* buf)
 
   CRegExp reTitle(true);
   reTitle.RegComp("StreamTitle=\'(.*?)\';");
-
+  
   if (reTitle.RegFind(strBuffer.c_str()) != -1)
   {
     std::string newtitle(reTitle.GetMatch(1));
-    result = (m_tag.GetTitle() != newtitle);
-    m_tag.SetTitle(newtitle);
+
+    CRegExp reParseTitle(true);
+    reParseTitle.RegComp("(?P<a>.*) - (?P<t>.*)");
```

Comment:
As I've written most radio stations a have seen so far are using this scheme. Other stations which split their station names with a '-' mostly doing this without spaces. Replace the regex if you think an other would fit better. I use this currently in an custom build of libreelec to display this information in a 4x20 display. The artist and song title get their own lines. If this will not go into Kodi i have to find an other way to extract this information afterwards. Nevertheless the station name should be published by kodi.